webapp/packages/plugin-connections/src/ConnectionForm/DriverProperties/DriverProperties.tsx

```diff
@@ -41,7 +42,20 @@ export const DriverProperties: TabContainerPanelComponent<IConnectionFormProps>
       propertiesList.splice(propertiesList.indexOf(property), 1);
     }
 
-    return { propertiesList, add, remove };
+    function reset() {
+      propertiesList.splice(0, propertiesList.length);
+    }
+
+    return { propertiesList, add, remove, reset };
+  });
+
+  useExecutor({
+    executor: optionsPart.onDriverIdChange,
+    handlers: [
+      function handleDriverChange() {
+        state.reset();
+      },
+    ],
```

Comment:
This may not work if the component is unmounted (when the options tab is active)

Reply:
We dont have access to this state in part, it exists only on the component level